### DISCUSSION TITLE
Is it Possible to Control the Display/Non-Display of Wallet App Icons in "View All" When Connecting via Wallet Connect?

### DISCUSSION THREAD

**Discussion by manakobayakawa** [WalletConnect/walletconnect-monorepo#4135] [Developer Support] [answered]
Dear Team,

I am in the process of developing a wallet app and am interested in restricting the range of external sites it can connect to. I have a few questions regarding this, especially in the context of using Wallet Connect with external sites like Uniswap and Opensea.

1. Controlling Visibility in Wallet Icon List: Is it possible to manage the display or non-display of our wallet app in the "View All" wallet icon list when connecting to external sites via Wallet Connect? Although I assume that this might be technically feasible on the Dapps' side, I wonder if it's possible to have our wallet app appear in the list only when connecting to specific sites. If such control is achievable, could you please outline the necessary procedures, including application conditions, required items, application flow, time frame for the review process, and the necessary communication channels?

2.  Procedures for Changing Visibility: What steps need to be taken to modify the visibility (display or non-display) of our wallet app in the list?

I appreciate your assistance with these queries and look forward to a prompt response.

Best regards,

Mana

**Comment by glitch-txs** [comment, marked as answer]
Hi, you can filter the origin of the website using Verify API: https://docs.walletconnect.com/web3wallet/verify
For your wallet to be visible in our modals you can submit your project in the Cloud website: https://docs.walletconnect.com/cloud/explorer-submission

**Reply by manakobayakawa** [reply]
Thank you for your reply!
From looking at the Verify API: https://docs.walletconnect.com/web3wallet/verify, it appears possible to exclude connections with domains that are not listed in WalletConnect’s domain registry. Is it possible for wallet providers to customize this domain registry? 
(In other words, is it possible to set up the wallet in such a way that it can only connect with one specific domain?)

**Reply by glitch-txs** [reply]
Yes, Iif you look at the code you can get the origin of the website, you can then filter by rejecting or not the connection
